### PR TITLE
Export cancer's Coverage and QC report with PDFKit

### LINKED ISSUE #3119: PDF "Coverage and QC report" from cancer cases doesn't fit the A4 page even in landscape layout
Check [this case](https://scout.scilifelab.se/cust000/NEQAS-somaticWES) suggested by @moahaegglund. 

I've reproduced it locally by:
- Going to the HTML report
- Viewing the source code
- Pasting the code into the "Coverage and QC report" of the demo cancer sample
- Loading this demo cancer sample locally

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug:
- Updated the `Coverage and QC` file for the demo cancer sample (note that the export of this exact file in main branch creates a chipped PDF file)
- Fixes #3119 by replacing the lib used to export the report (PDFKit instead of Weasyprint)
- Fixes the missing `Coverage and QC` HTML report for demo cancer case

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh your.email@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. Check this case on scout stage (main branch): https://scout-stage.scilifelab.se/cust000/ACC7948A21_ds
2. Install this branch on cg-vm1 and make sure that both HTML and PDF reports work as they should for the given sample

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/utils.py [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
 
 from bson.objectid import ObjectId
 from flask import abort, flash, render_template, request
 from flask_login import current_user
+from pdfkit import from_string
 
 LOG = logging.getLogger(__name__)
 
 
+def html_2_pdf_file(html_file_path, orientation, dpi=1000):
```
Comment (northwestwitch) [inline scout/server/utils.py:18]: If it nicely we can reuse this function to export any PDF from HTML in Scout!

**Comment by northwestwitch** [comment]
This doesn't work via docker yet (works on docker on my local machine though!). The problem is probably the fact that PDFKit doesn't find the executable of wkhtmltopdf, check here: https://github.com/JazzCore/python-pdfkit/issues/133

**Comment by northwestwitch** [comment]
This works with conda but when it comes to run via Docker then there is a dependency (wkhtmltopdf) that pdfkit doesn't find, even if it's installed. I'm starting to dislike PDFKit..

**Comment by dnil** [comment]
Doesn't sound all that encouraging. 😞 I don't have a lot of experience mixing npm js and python in several layers, but one would kind of expect containers to be a good bet on how to make it work. One could always fire up conda within the container... 😆

**Comment by northwestwitch** [comment]
> I don't have a lot of experience mixing npm js and python in several layers,

PDFkit is a wrapper around a [wkhtmltopdf](https://wkhtmltopdf.org/). There is no npm js involved, but that doesn't simplify the situation 😆  
I'm going to give it another try today.. The solution has potential!

**Comment by dnil** [comment]
Ah, there is a pypi python pdfkit as well! PDFKit is also a js wrapper/package with npm as the primary install obviously. 😊

**Comment by northwestwitch** [comment]
Today, with a fresh mind I understood what is wrong with that test! Since you need to install that package ( wkhtmltopdf) manually, the github action fails because it's never installed in there. The test should pass if I mock the command!

**Comment by northwestwitch** [comment]
This is ready for review now! 🥳

File: scout/server/blueprints/cases/views.py [hunk 2]
```diff
@@ -741,33 +741,45 @@ def gene_fusion_report(institute_id, case_name, report_type):
...
+    if coverage_qc_report is None:
         return abort(404)
 
-    coverage_qc_report = data["case"]["coverage_qc_report"]
     report_format = request.args.get("format", "html")
+
+    out_dir = os.path.abspath(os.path.dirname(coverage_qc_report))
+    filename = os.path.basename(coverage_qc_report)
+
+    LOG.warning(out_dir)
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/views.py:754]:
```suggestion
```

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html [hunk 3]
```diff
@@ -565,11 +565,11 @@ <h5 class="modal-title" id="archiveCaseLabel">Confirm archive case</h5>
     </div>
     <div>
       <a href="{{ url_for('cases.coverage_qc_report', institute_id=institute,
-                        case_name=case.display_name, date=date) }}" target="_blank"><i class="fa fa-link"></i></a>
+                        case_name=case.display_name) }}" target="_blank"><i class="fa fa-link"></i></a>
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:568]: Seems unused. May have been an idea there to append multiple date stamped ones, like for the delivery report.

File: scout/server/blueprints/cases/views.py [hunk 4]
```diff
@@ -741,33 +741,43 @@ def gene_fusion_report(institute_id, case_name, report_type):
 def coverage_qc_report(institute_id, case_name):
     """Display coverage and qc report."""
     _, case_obj = institute_and_case(store, institute_id, case_name)
-    data = controllers.multiqc(store, institute_id, case_name)
-    if data["case"].get("coverage_qc_report") is None:
+    coverage_qc_report = case_obj.get("coverage_qc_report")
```
Comment (dnil) [inline scout/server/blueprints/cases/views.py:744]: 🧐 👍

File: scout/server/utils.py [hunk 5]
```diff
@@ -18,7 +18,7 @@
 def html_to_pdf_file(html_string, orientation, dpi=96):
     """Creates a pdf file from the content of an HTML file
     Args:
-        html_string(string): An HTML string to be rendered as PDF
+        html_string(string): an HTML string to be rendered as PDF
```
Comment (dnil) [inline scout/server/utils.py:21]: 👍

File: tests/server/blueprints/cases/test_cases_views.py [hunk 6]
```diff
@@ -778,6 +778,31 @@ def test_pdf_delivery_report(app, institute_obj, case_obj, user_obj):
         assert resp.mimetype == "application/pdf"
 
 
+def test_pdf_coverage_qc_report(app, institute_obj, cancer_case_obj, user_obj):
```
Comment (dnil) [inline tests/server/blueprints/cases/test_cases_views.py:781]: 🥇

File: tests/server/test_server_utils.py [hunk 7]
```diff
@@ -1,11 +1,32 @@
...
 
 import pytest
 from flask import url_for
 
 from scout.server.links import get_variant_links
-from scout.server.utils import append_safe, find_index, variant_case
+from scout.server.utils import append_safe, find_index, html_to_pdf_file, variant_case
+
+
+def test_html_to_pdf_file():
```
Comment (dnil) [inline tests/server/test_server_utils.py:12]: 🥇

File: scout/server/blueprints/cases/views.py [hunk 8]
```diff
@@ -741,33 +741,43 @@ def gene_fusion_report(institute_id, case_name, report_type):
...
-                HTML(string=source_code),
-                download_filename=case_obj["display_name"]
-                + "_"
-                + datetime.datetime.now().strftime("%Y-%m-%d")
-                + "_coverage_qc_report.pdf",
-            )
+        try:
+            with open(os.path.abspath(coverage_qc_report), "r") as html_file:
+                source_code = html_file.read()
+                bytes_file = html_to_pdf_file(source_code, "landscape", 300)
```
Comment (dnil) [inline scout/server/blueprints/cases/views.py:758]: 💯

**Review by dnil — APPROVED** [review]

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html (hunk 3 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:568]: Perhaps!

**Pull request merged**